File: Lab6/src/main/java/com/example/lab6/data/Repository.java
package com.example.lab6.data;

import java.util.List;

public interface Repository {
    List<Order> getAll();
    Order getById(int id);
    List<Order> getAllByDeliveryAddress(String deliveryAddress);

    List<Order> getAllByPrepayment(boolean prepayment);

    boolean addOrder(Order order);
    boolean updateOrder(int id, Order order);
    boolean deleteOrder(int id);
}

File: Lab6Kotlin/src/main/kotlin/com/example/lab6/data/Repository.kt
package com.example.lab6.data

interface Repository {
    val all: List<Order>
    fun getById(id: Int): Order
    fun getAllByDeliveryAddress(deliveryAddress: String): List<Order>
    fun getAllByPrepayment(prepayment: Boolean): List<Order>
    fun addOrder(order: Order): Boolean
    fun updateOrder(id: Int, order: Order): Boolean
    fun deleteOrder(id: Int): Boolean
}